### PR TITLE
refactor transport popup

### PR DESCRIPTION
**Pull request by christianguenter2**
first try refactor transport popup
#1333

### CODE REVIEW

**Comment by larshp** [comment]
hi, I'll add comments in #1333

**Comment by larshp** [comment]
for reading package data, we should introduce a method in class `zcl_abapgit_sap_package`

File: src/ui/zcl_abapgit_services_repo.clas.abap [hunk 1]
```diff
@@ -152,6 +152,14 @@ CLASS ZCL_ABAPGIT_SERVICES_REPO IMPLEMENTATION.
         RETURN.
     ENDTRY.
 
+    IF ls_checks-transport-required = abap_true.
+
+      zcl_abapgit_popups=>popup_transport_request(
+        IMPORTING
+          ev_order = ls_checks-transport-transport ).
+
+    ENDIF.
```
Comment (larshp) [inline src/ui/zcl_abapgit_services_repo.clas.abap]: move inside the CATCH zcx_abapgit_cancel block

File: src/ui/zcl_abapgit_popups.clas.abap [hunk 2]
```diff
@@ -151,6 +151,13 @@ CLASS zcl_abapgit_popups DEFINITION
         !cv_show_popup TYPE char01
       RAISING
         zcx_abapgit_exception .
+    CLASS-METHODS popup_transport_request
+      EXPORTING
+        !ev_order TYPE trkorr
+        !ev_task  TYPE trkorr
```
Comment (larshp) [inline src/ui/zcl_abapgit_popups.clas.abap]: are both order and task needed? can probably be changed to 1 RETURNING value

File: src/ui/zcl_abapgit_popups.clas.abap [hunk 3]
```diff
@@ -1428,4 +1435,36 @@ CLASS ZCL_ABAPGIT_POPUPS IMPLEMENTATION.
...
+      EXCEPTIONS
+        no_correction_selected = 1
+        display_mode           = 2
+        object_append_error    = 3
+        recursive_call         = 4
+        wrong_order_type       = 5
+        OTHERS                 = 6.
+
+    IF sy-subrc = 1.
+      zcx_abapgit_exception=>raise( 'cancelled' ).
```
Comment (larshp) [inline src/ui/zcl_abapgit_popups.clas.abap]: zcx_abapgit_cancel

**Comment by christianguenter2** [comment]
fixed

**Pull request merged**